--- deps/ntlmclient/compat.h.orig	2020-11-16 14:33:42.000000000 +0100
+++ deps/ntlmclient/compat.h	2020-11-16 14:33:44.000000000 +0100
@@ -21,7 +21,11 @@
 # include <stdbool.h>
 #endif
 
-#if defined(_WIN32) || defined(__APPLE__)
+#if defined(__APPLE__) && !defined(htonll)
+# include <machine/endian.h>
+# include <libkern/OSByteOrder.h>
+# define htonll OSSwapInt64
+#elif defined(_WIN32) || defined(__APPLE__)
 /* winsock and macOS > 10.9 have htonll already */
 #elif defined(__linux__)
 /* See man page endian(3) */
