From d7d6f7456950f26a0d72550bbb3c5fb7797bb05f Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sun, 27 Sep 2020 15:18:46 +0200
Subject: [PATCH] macOS: Define htonll as OSSwapInt64

---
 deps/ntlmclient/compat.h | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/deps/ntlmclient/compat.h b/deps/ntlmclient/compat.h
index 6e5ef6466..bc6a09b93 100644
--- a/deps/ntlmclient/compat.h
+++ b/deps/ntlmclient/compat.h
@@ -21,8 +21,13 @@
 # include <stdbool.h>
 #endif
 
-#if defined(_WIN32) || defined(__APPLE__)
-/* winsock and macOS > 10.9 have htonll already */
+#if defined(_WIN32)
+/* winsock has hton11 already */
+#elif defined(__APPLE__) && !defined(htonll)
+/* macOS does not */
+# include <machine/endian.h>
+# include <libkern/OSByteOrder.h>
+# define htonll OSSwapInt64
 #elif defined(__linux__)
 /* See man page endian(3) */
 # include <endian.h>
-- 
2.23.0

