diff --git a/deps/ntlmclient/compat.h b/deps/ntlmclient/compat.h
index efdf34514..4394f537c 100644
--- a/deps/ntlmclient/compat.h
+++ b/deps/ntlmclient/compat.h
@@ -24,6 +24,10 @@
 #ifdef __linux__
 # include <endian.h>
 # define htonll htobe64
+#elif defined(__APPLE__) && !defined(htonll)
+# include <machine/endian.h>
+# include <libkern/OSByteOrder.h>
+# define htonll OSSwapInt64
 #elif defined(__OpenBSD__)
 /* See man page htobe64(3) */
 # include <endian.h>
